nilearn/glm/contrasts.py

```diff
@@ -525,15 +527,30 @@ def compute_fixed_effects(
             f"from the number of variance images ({len(variance_imgs)})."
         )
 
-    if isinstance(mask, NiftiMasker):
+    if isinstance(mask, (NiftiMasker, SurfaceMasker)):
         masker = mask.fit()
     elif mask is None:
-        masker = NiftiMasker().fit(contrast_imgs)
+        if isinstance(contrast_imgs[0], SurfaceImage):
+            surf_data = {}
+            for part in contrast_imgs[0].mesh.parts:
+                surf_data[part] = np.ones(
+                    contrast_imgs[0].data.parts[part].shape[0], dtype=bool
+                )
+            mask_img = SurfaceImage(mesh=contrast_imgs[0].mesh, data=surf_data)
+            masker = SurfaceMasker(mask_img=mask_img).fit(contrast_imgs[0])
```

Comment:
Can we not jsut do ?

```
            masker = SurfaceMasker().fit(contrast_imgs[0])
```

the fit of the surface masker does implicitly build a mask if none is passed at instanciation:

https://github.com/nilearn/nilearn/blob/e2c7e3c203f22e1cc40d5678e94dc4b3421c7101/nilearn/experimental/surface/maskers.py#L152

Reply:
I don't understand: in the case `mask_img` is provided, we should actually use it.

Reply:
agreed
my comment was unclear: let me try again